bluetooth/src/androidLibMain/kotlin/scanner/Scanner.kt

```diff
@@ -226,4 +226,10 @@ actual class Scanner internal constructor(
             } else null
         )
     }
+
+    // TODO: Filter devices by given [withServices]
+    override fun pairedDevices(withServices: Set<UUID>) = bluetoothAdapter
+        ?.bondedDevices
+        ?.map { it.address }
+        ?: emptyList()
 }
```

Comment:
`BluetoothDevice` exposes `uuids` property that returns an array of `UUID`s representing each server blt device service's id. If we get that we can filter devices by checking whether this array of services contains a uuid that is also inside `withServices`.

```suggestion
    override fun pairedDevices(withServices: Set<UUID>) = bluetoothAdapter
        ?.bondedDevices
        ?.filter { it.uuids.any { it.uuid in withServices } }
        ?: emptyList()
}
```

Reply:
e564e6eb